dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceConfiguration.java

```diff
@@ -630,6 +630,7 @@ public static class ElasticSearchConfig {
         private String user;
         private String password;
         private Integer maxConcurrentSessions;
+        private boolean isIndexOnStart = false;
```

Comment:
is this needed with the lock approach?

Reply:
This was implemented per Steve's suggestion https://github.com/dockstore/dockstore-deploy/pull/560#discussion_r947075759. 

Turns out it's also needed for a lot of tests to pass without making changes to the test files. In a lot of tests, the Elasticsearch client is not started, so the attempt to check if the mappings exist throws a NPE. With this flag being set as false by default, the tests won't even execute this Elasticsearch code.

Reply:
Oh, this is a bit inconsistent with our general strategy in ElasticListener (same deal with discourse integration). 
https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java#L110 and https://github.com/dockstore/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java#L143
log errors but do not throw exceptions when elastic search is not available